consumer-test/TestRunnerThread.groovy

```diff
@@ -0,0 +1,136 @@
+@Grab('org.yaml:snakeyaml:1.17')
+
+import org.yaml.snakeyaml.Yaml
+
+class TestRunnerThread extends Thread {
+
+    static def workspacesRootDir
+    static def libraryVersionUnderTest
+    static def repositoryUnderTest
+
+    Process currentProcess
+    StringBuilder stdOut = new StringBuilder()
+    StringBuilder stdErr = new StringBuilder()
+    int lastPrintedStdOutLine = -1
+    public def returnCode = -1
+    public def lastCommand
+    def area
+    def testCase
+    def uniqueName
+    def testCaseRootDir
+    def testCaseWorkspace
+    def testCaseConfig
+
+    TestRunnerThread(testCaseFilePath) {
+        // Regex pattern expects a folder structure such as '/rootDir/areaDir/testCase.extension'
+        def testCaseMatches = (testCaseFilePath.toString() =~
+            /^[\w\-]+\\/([\w\-]+)\\/([\w\-]+)\..*\u0024/)
+        this.area = testCaseMatches[0][1]
+        this.testCase = testCaseMatches[0][2]
+        if (!area || !testCase) {
+            throw new RuntimeException("Expecting file structure '/rootDir/areaDir/testCase.yml' " +
+                "but got '${testCaseFilePath.toString()}'.")
+        }
+        this.uniqueName = "${area}|${testCase}"
+        this.testCaseRootDir = new File("${workspacesRootDir}/${area}/${testCase}")
+        this.testCaseWorkspace = "${testCaseRootDir}/workspace"
+        this.testCaseConfig = new Yaml().load((testCaseFilePath as File).text)
+    }
+
+    void run() {
+        println "[INFO] Test case '${uniqueName}' launched."
+
+        if (!testCaseRootDir.mkdirs()) {
```

Comment:
[paranoia] what about a check if the directory does already exist. We get also `false` in case the directory is already present.

Reply:
The sequence we have currently does not allow this, because in the controller the parent directory is deleted [line 25](https://github.com/SAP/jenkins-library/blob/3693aa2b213eff5c0cd622c7cb2d70b065fc0e9e/consumer-test/consumerTestController.groovy#L25).